src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedPhysicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,354 @@
+			printOperatorInfoForOpGlobalToLocal( (PhysicalOpGlobalToLocal) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpHashJoin ) {
+			printOperatorInfoForOpHashJoin( (PhysicalOpHashJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpHashRJoin ) {
+			printOperatorInfoForOpHashRJoin( (PhysicalOpHashRJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpIndexNestedLoopsJoin ) {
+			printOperatorInfoForOpIndexNestedLoopsJoin( (PhysicalOpIndexNestedLoopsJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpLocalToGlobal ) {
+			printOperatorInfoForOpLocalToGlobal( (PhysicalOpLocalToGlobal) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpMultiwayUnion ) {
+			printOperatorInfoForMultiwayUnion( (PhysicalOpMultiwayUnion) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpNaiveNestedLoopsJoin ) {
+			printOperatorInfoForOpNaiveNestedLoopsJoin( (PhysicalOpNaiveNestedLoopsJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpParallelMultiLeftJoin ) {
+			printOperatorInfoForOpParallelMultiLeftJoin( (PhysicalOpParallelMultiLeftJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpRequest ) {
+			printOperatorInfoForRequest( (PhysicalOpRequest) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpSymmetricHashJoin ) {
+			printOperatorInfoForOpSymmetricHashJoin( (PhysicalOpSymmetricHashJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else {
+			throw new IllegalArgumentException( "Unexpected operator type: " + rootOp.getClass().getName() );
+		}
+		for ( int i = 0; i < plan.numberOfSubPlans(); ++i ) {
+			planWalk( plan.getSubPlan(i), i, planLevel+1, plan.numberOfSubPlans(), out, indentLevelString );
+		}
+	}
+	
+	protected void printOperatorInfoForBinaryUnion ( final PhysicalOpBinaryUnion op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "binaryUnion (" + op.getID() + ") " );
+		printLogicalOpForPhysicalOp( op.getLogicalOperator(), out, indentLevelString, indentLevelStringForOpDetail );
+	}
+	
+	protected void printOperatorInfoForBindJoin ( final PhysicalOpBindJoin op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "bindJoin (" + op.getID() + ") " );
+		printLogicalOpForPhysicalOp( op.getLogicalOperator(), out, indentLevelString, indentLevelStringForOpDetail );
+	}
```

Comment:
As `PhysicalOpBindJoin` is a physical operator for `LogicalOpTPAdd`, `LogicalOpTPOptAdd`, and the likes, it has a fed.member and a query pattern as well. Hence, the `- fm ...` and `- pattern ...` lines should be printed here as well.

Same comment for the other bind-join variations (`PhysicalOpBindJoinWithFILTER`, etc.)

Reply:
In `printLogicalOpForPhysicalOp`, I have a check on if the logical operator is `tpAdd`, `tpOptAdd`, `BGPAdd`, `BGPOptAdd`, `GPAdd`, `GPOptAdd`. Therefore `- fm ...` and `- pattern ...` lines will be printed.